src/tests/test_transform.py

```diff
@@ -2,15 +2,115 @@
 # This file may be used under the terms of the GNU General Public License, version 2 or later.
 # For more details see: https://www.gnu.org/licenses/gpl-2.0.html
 
+from src.transform.datastructures import AddonVersion, NVDAVersion, VersionCompatibility
+from src.transform.transform import getLatestAddons, _isAddonCompatible
+from src.tests.generateData import MockAddon
 import unittest
 
+V_2020_1 = NVDAVersion(2020, 1)
+V_2020_2 = NVDAVersion(2020, 2)
+V_2020_3 = NVDAVersion(2020, 3)
+V_2021_1 = NVDAVersion(2021, 1)
+V_2021_1_1 = NVDAVersion(2021, 1, 1)
+V_2021_2 = NVDAVersion(2021, 2)
+V_2022_1 = NVDAVersion(2022, 1)
+nvdaVersion2020_2 = VersionCompatibility(V_2020_2, V_2020_2, V_2020_1)
+nvdaVersion2020_3 = VersionCompatibility(V_2020_3, V_2020_3, V_2020_1)
+nvdaVersion2021_1 = VersionCompatibility(V_2021_1, V_2021_1, V_2021_1)
+nvdaVersion2021_1_1 = VersionCompatibility(V_2021_1_1, V_2021_1, V_2021_1)
+nvdaVersion2022_1 = VersionCompatibility(V_2022_1, V_2022_1, V_2022_1)
 
-class Test_testExample(unittest.TestCase):
-	def test_example_true(self):
-		self.assertTrue(True)
 
-	def test_example_pass(self):
-		pass
+class Test_isAddonCompatible(unittest.TestCase):
+	def test_valid_with_api(self):
+		"""Confirm an addon is compatible with a fully tested API"""
+		addon = MockAddon()
```

Comment:
What is the benefit of using a `MockAddon` over the real datatype in this case? Will it raise if you access an unset variable?

Typically I would use a mock to supply a specific value, or to spy on values given to the mock, or methods called.

Reply:
Please expand on the comment in `def MockAddon() -> Addon:` to explain.

Reply:
Done as of b40a61d